RedBeanPHP/QueryWriter/AQueryWriter.php

```diff
@@ -1348,16 +1340,19 @@ public function setUseCache( $yesNo )
 	 * Clears the internal query cache array and returns its overall
 	 * size.
 	 *
-	 * @return integer
+	 * @return mixed
 	 */
-	public function flushCache( $newMaxCacheSizePerType = NULL )
+	public function flushCache( $newMaxCacheSizePerType = NULL, $countCache = TRUE )
 	{
 		if ( !is_null( $newMaxCacheSizePerType ) && $newMaxCacheSizePerType > 0 ) {
 			$this->maxCacheSizePerType = $newMaxCacheSizePerType;
 		}
-		$count = count( $this->cache, COUNT_RECURSIVE );
+		if ( $countCache ) {
+			$count = count( $this->cache, COUNT_RECURSIVE );
+			$this->cache = array();
+			return $count;
+		}
```

Comment:
We could reduce code duplication even further with:

    $count = $countCache ? count( $this->cache, COUNT_RECURSIVE );
    $this->cache = array();
    return $cache;

Also, we then have just one return statement.

Reply:
Fixed